envelope.go

```diff
@@ -59,7 +127,9 @@ type envelopeWriter struct {
 
 func (w *envelopeWriter) Marshal(message any) *Error {
 	if message == nil {
-		if _, err := w.writer.Write(nil); err != nil {
+		// Send a zero-length message.
+		payload := bytes.NewReader(nil)
```

Comment:
Might be better to use a no-op empty struct type, so there's no allocation here and also to make it clear we aren't actually sending a message.

Reply:
Created the `nopPayload` type that implements `messagePayload`.